src/main/java/ch/njol/skript/lang/function/FunctionRegistry.java

```diff
@@ -0,0 +1,588 @@
+			.formatted(name, Arrays.toString(Arrays.stream(identifier.args).map(Class::getSimpleName).toArray()),
+				namespace));
+	}
+
+	/**
+	 * Checks if a signature with the given name and arguments exists in the given namespace.
+	 *
+	 * <ul>
+	 * <li>If {@code namespace} is null, only global signatures will be checked.</li>
+	 * <li>If {@code args} is null or empty,
+	 * the first function with the same name as the {@code name} param will be returned.</li>
+	 * </ul>
+	 *
+	 * @param namespace The namespace to check in.
+	 *                  Usually represents the path of the script a signature is registered in.
+	 * @param name      The name of the function.
+	 * @param args      The types of the arguments of the function.
+	 * @return True if a signature with the given name and argument types exists in the script, false otherwise.
+	 */
+	public static boolean signatureExists(@Nullable String namespace, @NotNull String name, Class<?>... args) {
+		if (namespace == null) {
+			return signatureExists(GLOBAL_NAMESPACE, FunctionIdentifier.of(name, false, args));
+		}
+
+		return signatureExists(new Namespace(Scope.LOCAL, namespace.toLowerCase()), FunctionIdentifier.of(name, true, args));
+	}
+
+	/**
+	 * Checks if a function with the given name and arguments exists in the namespace.
+	 *
+	 * @param namespace  The namespace to check in.
+	 * @param identifier The identifier of the function.
+	 * @return True if a function with the given name and arguments exists in the namespace, false otherwise.
+	 */
+	private static boolean signatureExists(@NotNull Namespace namespace, @NotNull FunctionIdentifier identifier) {
+		Preconditions.checkNotNull(namespace, "namespace is null");
+		Preconditions.checkNotNull(identifier, "identifier is null");
+
+		Map<String, Set<FunctionIdentifier>> javaIdentifiers = identifiers.getOrDefault(namespace, new HashMap<>());
+
+		if (!javaIdentifiers.containsKey(identifier.name)) {
+			return false;
+		}
+
+		for (FunctionIdentifier other : javaIdentifiers.get(identifier.name)) {
+			if (identifier.equals(other)) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Gets a function from a script. If no local function is found, checks for global functions.
+	 *
+	 * <ul>
+	 * <li>If {@code namespace} is null, only global functions will be checked.</li>
+	 * <li>If {@code args} is null or empty,
+	 * the first function with the same name as the {@code name} param will be returned.</li>
+	 * </ul>
+	 *
+	 * @param namespace The namespace to get the function from.
+	 *                  Usually represents the path of the script this function is registered in.
+	 * @param name      The name of the function.
+	 * @param args      The types of the arguments of the function.
+	 * @return The function with the given name and argument types, or null if no such function exists.
+	 */
+	public static Function<?> function(@Nullable String namespace, @NotNull String name, Class<?>... args) {
```

Comment:
I don't like this naming scheme here. With things like records, `.xyz()` is get, `.xyz(something)` is set. And especially with `function` meaning to do something, I would like this to be clear that it is getting a function and not doing anything else. Same with `signature(...)`